### PR TITLE
[TERRA-537] Split test.ControlledResourceFixtures

### PR DESCRIPTION
**Pull request by dexamundsen**
Current set up for test fixtures - 
- class ControlledResourceFixtures has Azure, GCP and flex (non-cloud specific) resource fixtures

Changes - 
- Split the large file into cloud specific classes
- Non-cloud specific functions and constants are retained in the same file
- Consolidate Azure batch pool functions and constants into ControlledAzureResourceFixtures 
-----
Other - remove unnecessary 'final' keyword only in affected files
No functional changes. No tests are changed

-----

Putting out this LARGE changeset as is for 2 reasons
- All changes are refactor / rename changes in tests. Not changes in any of main functional files. This avoids the risk of functional regressions
- Refactor / package changes typically touch large files. Pulling it off all at once to avoid frequent code churns

### CODE REVIEW

**Comment by dexamundsen** [comment]
@ddietterich If you have more PRs in the pipeline, I can pause this

**Review by zloery — APPROVED** [review]

File: service/src/test/java/bio/terra/workspace/amalgam/landingzone/azure/LandingZoneApiControllerTest.java [hunk 1]
```diff
@@ -1,6 +1,6 @@
 package bio.terra.workspace.amalgam.landingzone.azure;
 
-import static bio.terra.workspace.common.utils.MockMvcUtils.AUTH_HEADER;
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply service/src/test/java/bio/terra/workspace/amalgam/landingzone/azure/LandingZoneApiControllerTest.java:3]: Renamed them solely to separate out utils meant only for test, and tests meant for app utils.

File: service/src/test/java/bio/terra/workspace/app/controller/ControlledGcpResourceApiControllerBqDatasetConnectedTest.java [hunk 2]
```diff
@@ -187,25 +186,22 @@ public void setup() throws Exception {
    */
   @AfterEach
   public void resetFlightDebugInfo() {
-    AuthenticatedUserRequest defaultUserRequest =
-        userAccessUtils.defaultUser().getAuthenticatedRequest();
+    AuthenticatedUserRequest defaultUserRequest = userAccessTestUtils.defaultUserAuthRequest();
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply service/src/test/java/bio/terra/workspace/app/controller/ControlledGcpResourceApiControllerBqDatasetConnectedTest.java:189]: reverting in the interest of change set size

File: service/src/test/java/bio/terra/workspace/app/controller/ControlledFlexibleResourceApiControllerConnectedTest.java [hunk 3]
```diff
@@ -48,9 +48,9 @@ public class ControlledFlexibleResourceApiControllerConnectedTest extends BaseCo
   @Autowired MockMvc mockMvc;
   @Autowired MockMvcUtils mockMvcUtils;
   @Autowired ObjectMapper objectMapper;
-  @Autowired UserAccessUtils userAccessUtils;
+  @Autowired UserAccessTestUtils userAccessTestUtils;
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply service/src/test/java/bio/terra/workspace/app/controller/ControlledFlexibleResourceApiControllerConnectedTest.java:51]: > Out of the 9 files I've reviewed so far, 7 of them have only these changes:
> 
> * rename utils to testutils
> * rename fixtures to testfixtures
> * rename UserAccessUtils to UserAccessTestUtils
> 
> I don't think any of those changes are needed. None of them are related to restructuring the test fixtures. They are all cosmetic and, in my view, low value.
> 
> I'd like to see a PR that does the test refactoring without the cosmetic renames.

-- 
Yes, the cosmetic changes are renames / package moves

- Renamed packages solely to separate out utils meant only for test, and tests meant for app utils.
- renamed field for consistency

reverting them in the interest of change set size

File: service/src/test/java/bio/terra/workspace/amalgam/landingzone/azure/LandingZoneApiControllerTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (yuhuyoyo) [reply service/src/test/java/bio/terra/workspace/amalgam/landingzone/azure/LandingZoneApiControllerTest.java:3]: java file structure already separate out test and main tho

**Review by yuhuyoyo — APPROVED** [review]

**Comment by dexamundsen** [comment]
@yuhuyoyo - yea but within the test package there are utils for tests and tests for utils in the same package!

File: service/src/test/java/bio/terra/workspace/app/controller/ControlledAzureResourceApiControllerBatchPoolTest.java [hunk 4]
```diff
@@ -59,11 +60,9 @@ public void createBatchPool400WithNoParameters() throws Exception {
 
   @Test
   public void createBatchPool400WithInconsistentUAMI() throws Exception {
-    final ApiControlledResourceCommonFields commonFields =
-        ControlledResourceFixtures.makeDefaultControlledResourceFieldsApi();
+    final ApiControlledResourceCommonFields commonFields = makeDefaultControlledResourceFieldsApi();
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply service/src/test/java/bio/terra/workspace/app/controller/ControlledAzureResourceApiControllerBatchPoolTest.java]: done, only in files already being updated

File: service/src/test/java/bio/terra/workspace/amalgam/landingzone/azure/LandingZoneApiControllerTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (dexamundsen) [reply service/src/test/java/bio/terra/workspace/amalgam/landingzone/azure/LandingZoneApiControllerTest.java]: but within the test package there are utils for tests and tests for utils in the same package!

**Review by ddietterich — APPROVED** [review]

**Pull request merged**